tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.py

```diff
@@ -9,41 +9,52 @@
 ## to two Z bosons and subsequently to four leptons is called the "golden channel". The selection leads
 ## to a narrow invariant mass peak on top a relatively smooth and small background, revealing
 ## the Higgs at 125 GeV.
-##
-## The analysis is translated to a RDataFrame workflow processing about 300 MB of simulated events and data.
+## Systematic errors for the MC scale factors are computed and the Vary function of RDataFrame is used for plotting. 
+## The analysis is translated to an RDataFrame workflow processing about 300 MB of simulated events and data.
 ##
 ## \macro_image
 ## \macro_code
 ## \macro_output
 ##
 ## \date March 2020
 ## \author Stefan Wunsch (KIT, CERN)
-## \date May 2023
+## \date July 2022
+## \author Julia Mathe (CERN)
+## \date August 2023
 ## \author Marta Czurylo (CERN)
 
 import ROOT
 import os
 
-# Create the RDataFrame from the spec json file. The df106_HiggsToFourLeptons_spec.json is provided in the same folder as this tutorial
+# Enable Multi-threaded mode
+ROOT.EnableImplicitMT()
 
+# Create the RDataFrame from the spec json file. The df106_HiggsToFourLeptons_spec.json is provided in the same folder as this tutorial
 my_spec = os.path.join(ROOT.gROOT.GetTutorialsDir(), "dataframe", "df106_HiggsToFourLeptons_spec.json")
-
 df = ROOT.RDF.Experimental.FromSpec(my_spec) # Creates a single dataframe for all the samples
 
-# Access metadata information that is stored in the JSON config file of the RDataFrame
-# The metadata contained in the JSON file is accessible within a `DefinePerSample` call, through the `RDFSampleInfo` class
-
 # Add the ProgressBar feature
-
 ROOT.RDF.Experimental.AddProgressbar(df) 
 
+# Access metadata information that is stored in the JSON config file of the RDataFrame
+# The metadata contained in the JSON file is accessible within a `DefinePerSample` call, through the `RDFSampleInfo` class
 df = df.DefinePerSample("xsecs", 'rdfsampleinfo_.GetD("xsecs")')
-df = df.DefinePerSample("lumi", 'rdfsampleinfo_.GetD("lumi")') 
+df = df.DefinePerSample("lumi", 'rdfsampleinfo_.GetD("lumi")')
 df = df.DefinePerSample("sumws", 'rdfsampleinfo_.GetD("sumws")')
 df = df.DefinePerSample("sample_category", 'rdfsampleinfo_.GetS("sample_category")')
 
-# Select events for the analysis
+# We must further apply an MC correction for the ZZ decay due to missing gg->ZZ processes.
+ROOT.gInterpreter.Declare(
+"""
+float scale(unsigned int slot, const ROOT::RDF::RSampleInfo &id){
+                if (id.Contains("mc_363490.llll.4lep.root")){return 1.3f;}
+                else {return 1.0f;}
+}
```

Comment:
This should better be formatted with ROOTs [clang-format style](https://github.com/root-project/root/blob/master/.clang-format) as well.